github/lib/configure_repo.rb

```diff
@@ -147,4 +149,12 @@ def github_actions_test_job_name
       test_job.fetch("name", "test")
     end
   end
+  
+  def enable_vulnerability_alerts
+    client.enable_vulnerability_alerts(repo[:full_name])
+  end
+
+  def enable_automated_security_fixes
+    client.put("https://api.github.com/repos/#{repo[:full_name]}/automated-security-fixes", accept: "application/vnd.github+json")
```

Comment:
Strange that there is an official SDK method for [enable_vulnerability_alerts](https://www.rubydoc.info/gems/octokit/Octokit/Client/Repositories#enable_vulnerability_alerts-instance_method), but there isn't one for enabling Dependabot to raise security PRs?

I'm curious as to how you found this API method. From a quick Google search, I can't find anything on it, which makes me question how well supported it is and how long it may continue to work for.

Also just wanted to check if you've run these methods against a GOV.UK repo and observed that they work? Would be good to have before/after screenshots in the PR description so it's clearer what this PR intends to do.

Reply:
It's [this](https://github.com/octokit/octokit.rb/blob/main/lib/octokit/connection.rb#L36) method and there's [this in the docs](https://octokit.github.io/octokit.rb/Octokit/Connection.html#put-instance_method). It's doing the [same call in the background](https://github.com/octokit/octokit.rb/blob/main/lib/octokit/connection.rb#L147) as the other method, it's just that the other one was [a contribution by someone who forked the repo](https://github.com/octokit/octokit.rb/pull/1308).

I was planning to test this branch after getting a review, I think I can do that in Jenkins

Reply:
Thanks, useful to know about that fork contribution! Hmm I don't see any reference to `automated-security-fixes` in any of the links you provided, so still not sure where you found that endpoint?

By testing the branch in Jenkins, do you mean deploying this branch and then letting it run against all GOV.UK repos? It would be safer to do an isolated test first, i.e. reproduce the PUT calls locally and run against a chosen GOV.UK repo to see if it works. 🤞 Assuming the calls work, the code LGTM!

Reply:
If you mean the GitHub endpoint, it's documented [here](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#enable-automated-security-fixes). I'll try to test it locally on a repo

Reply:
That's the one! Thanks

Reply:
Tested on my machine on a repo and confirmed this works